<?xml version="1.0"?>
<launch>


	<!-- Arguments to set the MIA COM number  -->
	<arg name="Mia_COM_" default='0' doc="Int number of the Mia hand COM port" />
	<arg name="Mia_fs_" default='100' doc="Hz, frequency of the control loop of the Mia Hw interface" />
	<arg name="robotNamespace"         default="mia_hand_hw"/> <!-- /mia_hand -->

<group ns="$(arg robotNamespace)">


  <!-- Argument to select the type traj controller -->
	<arg name="tc_type" default="vel"  doc="type of trajectory controller to launch. Values: vel, pos "/>


	<!-- Load urd as robot description -->
	<param name="robot_description" command="$(find xacro)/xacro --inorder '$(find mia_hand_description)/urdf/mia_hand_launch.urdf.xacro' " />

<!--remap from="joint_states" to="pippo/joint_states" /-->

	<!-- convert joint states to TF transforms for rviz, etc -->
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false"/>

	<!-- Load joint controller configurations from YAML file to parameter server -->
	<rosparam file="$(find mia_hand_ros_control)/config/mia_hand_hw_controllers.yaml" command="load"/>
	<rosparam file="$(find mia_hand_ros_control)/config/mia_hand_hw_trajectory_controllers.yaml" command="load"/>


	<!-- Start the control node -->
	<node name="Mia_hw_node" pkg="mia_hand_ros_control" type="Mia_hw_node" output="screen" >
		<param name="Mia_COM_" type="int" value="$(arg Mia_COM_)"/> <!-- private param so as to match to the namespace used-->
		<param name="Mia_fs_" type="int" value="$(arg Mia_fs_)"/> <!-- private param so as to match to the namespace used-->
	</node>

	<!-- spawn controller -->
	<node name="trajectory_ctrl_spawner" pkg="controller_manager" type="spawner"
		output="screen"  args="mia_hand_hw_$(arg tc_type)_trajectory_controller" />

	<node pkg="controller_manager" type="spawner" name="joint_state_controller_spawner"
		output="screen"   args="joint_state_controller" />


	<!-- Load gui to teleop -->
	<node name="rqt_joint_trajectory_controller" pkg="rqt_joint_trajectory_controller" type="rqt_joint_trajectory_controller" />
</group>

</launch>
